test/BackendApi.Test/Controllers/ParcelControllerTest.cs

```diff
@@ -0,0 +1,252 @@
+            Model.Parcel[] actualParcels = Assert.IsType<Model.Parcel[]>(actionResult.Value);
+            Assert.Equal(new Model.Parcel[] { new Model.Parcel(_expectedParcel) }, actualParcels);
+        }
+
+        [Fact]
+        public void GetMyParcels_GetMultiple()
+        {
+            // Arrange
+            Entity.Parcel[] testParcels = getTestParcels(_expectedParcel);
+            _dbContext.Parcels.AddRange(testParcels);
+            _dbContext.SaveChanges();
+
+            // Act
+            var result = _parcelController.GetMyParcels(100, 100, 0, 0);
+
+            // Assert
+            JsonResult actionResult = Assert.IsType<JsonResult>(result);
+            Model.Parcel[] actualParcels = Assert.IsType<Model.Parcel[]>(actionResult.Value);
+            Model.Parcel[] expectedParcels = testParcels.Select(parcel => new Model.Parcel(parcel)).ToArray();
+            Assert.Equal(expectedParcels, actualParcels);
+        }
+
+        [Fact]
+        public void GetMyParcels_FilterAll()
+        {
+            // Arrange
+            Entity.Parcel[] testParcels = getTestParcels(_expectedParcel);
+            _dbContext.Parcels.AddRange(testParcels);
+            _dbContext.SaveChanges();
+
+            // Act
+            var result = _parcelController.GetMyParcels(0, 25, 10, 20);
+
+            // Assert
+            JsonResult actionResult = Assert.IsType<JsonResult>(result);
+            Model.Parcel[] actualParcels = Assert.IsType<Model.Parcel[]>(actionResult.Value);
+            Assert.Empty(actualParcels);
+        }
+
+        [Fact]
+        public void GetMyParcels_NoData()
+        {
+            // Act
+            var result = _parcelController.GetMyParcels();
+
+            // Assert
+            JsonResult actionResult = Assert.IsType<JsonResult>(result);
+            Model.Parcel[] actualParcels = Assert.IsType<Model.Parcel[]>(actionResult.Value);
+            Assert.Empty(actualParcels);
+        }
+        #endregion
+        #region GetMyParcelDetail
+        [Fact]
+        public void GetMyParcelDetail_NoData()
+        {
+            // Act
+            int expectedParcelId = 1;
+            var actualParcelDetail = _parcelController.GetMyParcels(expectedParcelId);
+
+            // Assert
+            Assert.IsType<NoContentResult>(actualParcelDetail);
+        }
+
+        [Fact]
+        public void GetMyParcelDetail_NonMatchingId()
+        {
+            // Arrange
+            Entity.Parcel expectedTestParcel = new Entity.Parcel
+            {
+                Id = 2
+            };
+            _dbContext.Parcels.Add(expectedTestParcel);
+            _dbContext.SaveChanges();
+
+            // Act
+            int expectedParcelId = 1;
+            var actualParcelDetail = _parcelController.GetMyParcels(expectedParcelId);
+
+            // Assert
+            Assert.IsType<NoContentResult>(actualParcelDetail);
+        }
+
+        [Fact]
+        public void GetMyParcelDetail_Matching()
```

Comment:
One test that I went back and forth on - a test to check that all of the fields supported by the database are returned by the API. I decided against it for now because I wasn't sure the test would have value due to the automapper, and this also may be more appropriate for an integration test, as I see a test like this being fragile to change.

Interested to hear other's thoughts.

Reply:
I'm generally against testing that all fields exist, Normally all fields are not required to be included in the API results, as the API can change the shape of the literal data based on how we want to present it.  Commonly this occurs when a role based access can't request certain data elements.

That being said, there are times I like tests like this if they are easily modified through templating/configuration.  In other words, there is some kind of logic that is showing/hiding properties.